apps/ledger-live-mobile/src/screens/ReceiveFunds/02-AddAccount.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { useEffect, useCallback, useState, useRef, memo } from "react";
+import { FlatList } from "react-native";
+import { concat, from } from "rxjs";
+import { ignoreElements } from "rxjs/operators";
+import { useDispatch } from "react-redux";
+import { useTranslation } from "react-i18next";
+import {
+  CryptoCurrency,
+  Account,
+  TokenAccount,
+  Currency,
+  AccountLike,
+  TokenCurrency,
+} from "@ledgerhq/live-common/lib/types";
+import { getCurrencyBridge } from "@ledgerhq/live-common/lib/bridge";
+import { Device } from "@ledgerhq/live-common/lib/hw/actions/types";
+
+import { Flex, InfiniteLoader } from "@ledgerhq/native-ui";
+import { makeEmptyTokenAccount } from "@ledgerhq/live-common/lib/account";
+import { replaceAccounts } from "../../actions/accounts";
+import logger from "../../logger";
+import { ScreenName } from "../../const";
+import { TrackScreen } from "../../analytics";
+import Button from "../../components/Button";
+import PreventNativeBack from "../../components/PreventNativeBack";
+import LText from "../../components/LText";
+import RetryButton from "../../components/RetryButton";
+import CancelButton from "../../components/CancelButton";
+import GenericErrorBottomModal from "../../components/GenericErrorBottomModal";
+import { prepareCurrency } from "../../bridge/cache";
+import AccountCard from "../../components/AccountCard";
+
+type RouteParams = {
+  currency: CryptoCurrency | TokenCurrency;
+  device: Device;
+  onSuccess?: (params?: any) => void;
+};
+
+type Props = {
+  navigation: any;
+  route: { params: RouteParams };
+  blacklistedTokenIds?: string[];
+  colors: any;
+};
+
+function AddAccountsAccounts({ navigation, route }: Props) {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const [scanning, setScanning] = useState(true);
+  const [error, setError] = useState(null);
+  const [scannedAccounts, setScannedAccounts] = useState<Account[]>([]);
+  const [cancelled, setCancelled] = useState(false);
+
+  const scanSubscription = useRef<any>();
+
+  const {
+    currency,
+    device: { deviceId },
+  } = route.params || {};
+
+  useEffect(() => {
+    startSubscription();
+    return () => stopSubscription(false);
+  }, []);
+
+  const startSubscription = useCallback(() => {
+    const c =
+      currency.type === "TokenCurrency" ? currency.parentCurrency : currency;
+    const bridge = getCurrencyBridge(c);
+    const syncConfig = {
+      paginationConfig: {
+        operation: 0,
+      },
+      blacklistedTokenIds: [],
+    };
+    // will be set to false if an existing account is found
+
+    scanSubscription.current = concat(
+      from(prepareCurrency(c)).pipe(ignoreElements()),
+      bridge.scanAccounts({
+        currency: c,
+        deviceId,
+        syncConfig,
+      }),
+    ).subscribe({
```

Comment:
i have a feeling this code is duplicated from somewhere else, and therefore we should code share more for maintainability.

Reply:
Added some comments to explain what is being done during scanning of accounts.
However maybe we can share it but the logic of handling the success stream of accounts is pretty specific here.